### DISCUSSION TITLE
Status Messages (Events)

### DISCUSSION THREAD

**Discussion by nireins** [ekutner/home-connect-hass#192] [General]
Hi, fist off all: very great work!

I use this integration for my Siemens Coffe Mashine EQ.9 and Neff Dishwascher.

I hope that I am correct at this point.

What I miss in this integration are the status messages, such as with the coffee machine: "Bean container empty", "Fill water tank", "Empty drip tray", etc. And for the dishwasher, something like: "Refill rinse aid", "Refill salt", etc. Such status messages are displayed in the Home Connect app. 

Is it at all possible to include this? 

Greetings

**Comment by Rep-Hueman** [comment]
That is also included under events! You see an event when the water tank is empty, you can make a script or template sensor for this

```
Template.yaml

  - trigger:
      - platform: event
        event_type: home_connect_alt_event
        event_data:
          key: ConsumerProducts.CoffeeMaker.Event.BeanContainerEmpty
          value: BSH.Common.EnumType.EventPresentState.Present
        id: "on"
      - platform: event
        event_type: home_connect_alt_event
        event_data:
          key: ConsumerProducts.CoffeeMaker.Event.BeanContainerEmpty
          value: BSH.Common.EnumType.EventPresentState.Off
        id: "off"
    binary_sensor:
      - name: espresso_boon_res_empty
        state: "{{ trigger.id }}"

**Comment by nireins** [comment]
Hi, thanks for that. 
However, I am literally on the tube right now :)

I tried to display the events in text via the following automation, but unfortunately it doesn't work.

alias: "Coffee Mashine: Events"
description: ""
trigger:
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: ConsumerProducts.CoffeeMaker.Event.BeanContainerEmpty
      value: BSH.Common.EnumType.EventPresentState.Present
    id: Bohnenbehälter leer
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: ConsumerProducts.CoffeeMaker.Event.BeanContainerEmpty
      value: BSH.Common.EnumType.EventPresentState.Off
    id: Bohnenbehälter leer aus
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: ConsumerProducts.CoffeeMaker.Event.WaterTankEmpty
      value: BSH.Common.EnumType.EventPresentState.Present
    id: Wassertank leer
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: ConsumerProducts.CoffeeMaker.Event.WaterTankEmpty
      value: BSH.Common.EnumType.EventPresentState.Off
    id: Wassertank leer aus
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: ConsumerProducts.CoffeeMaker.Event.DripTrayFull
      value: BSH.Common.EnumType.EventPresentState.Present
    id: Tropfschale leeren
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: ConsumerProducts.CoffeeMaker.Event.DripTrayFull
      value: BSH.Common.EnumType.EventPresentState.Off
    id: Tropfschale leeren aus
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: Dishcare.Dishwasher.Event.SaltNearlyEmpty
      value: BSH.Common.EnumType.EventPresentState.Present
    id: Salz nachfüllen
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: Dishcare.Dishwasher.Event.SaltNearlyEmpty
      value: BSH.Common.EnumType.EventPresentState.Off
    id: Salz nachfüllen aus
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: Dishcare.Dishwasher.Event.RinseAidNearlyEmpty
      value: BSH.Common.EnumType.EventPresentState.Present
    id: Klarspüler nachfüllen
  - platform: event
    event_type: home_connect_alt_event
    event_data:
      key: Dishcare.Dishwasher.Event.RinseAidNearlyEmpty
      value: BSH.Common.EnumType.EventPresentState.Off
    id: Klarspüler nachfüllen aus
condition: []
action:
  - choose:
      - conditions:
          - condition: trigger
            id: Bohnenbehälter leer
        sequence:
          - service: notify.mobile_app_flitze
            data:
              title: "***WARNUNG***"
              message: Bohnenbehälter ist leer
          - service: input_text.set_value
            data:
              value: Bohnenbehälter füllen!
            target:
              entity_id: input_text.events
      - conditions:
          - condition: trigger
            id: Wassertank leer
        sequence:
          - service: notify.mobile_app_flitze
            data:
              title: "***WARNUNG***"
              message: Wassertank ist leer
          - service: input_text.set_value
            data:
              value: Wassertank füllen!
            target:
              entity_id: input_text.events
      - conditions:
          - condition: trigger
            id: Tropfschale leeren
        sequence:
          - service: notify.mobile_app_flitze
            data:
              message: Tropfschale leeren!
              title: "***WARNUNG***"
          - service: input_text.set_value
            data:
              value: Tropfschale leeren!
            target:
              entity_id: input_text.events
      - conditions:
          - condition: trigger
            id: Klarspüler nachfüllen
        sequence:
          - service: input_text.set_value
            data:
              value: Klarspüler nachfüllen!
            target:
              entity_id: input_text.events
  - if:
      - condition: trigger
        id: Bohnenbehälter leer aus
      - condition: and
        conditions:
          - condition: trigger
            id: Wassertank leer aus
          - condition: trigger
            id: Tropfschale leeren aus
    then:
      - service: input_text.set_value
        data:
          value: " "
        target:
          entity_id: input_text.events
mode: single


Do I have something wrong understood?

Can I ask you once for your help, how the whole thing must look correct? Thank you very much.

**Comment by ekutner** [comment]
This is not an issue. Moving to discussion. Unfortunately I don't have an example ready to show and don't have the capacity to build one.

**Reply by nireins** [reply]
Hello, I am so sorry.... I made a big mistake... I corrected the error and now it works. 

Thanks again for the tip with the template.

**Reply by d0np3p3** [reply]
Hey could you post your final templates?
Maybe it could be added to the readme, or we can make an template_example.md

**Reply by steve-hodgkiss** [reply]
https://api-docs.home-connect.com/events/#home-appliance-events lists all (most?) of the status events generated by appliances. e.g. dishwasher salt being empty.
Was wondering about the best way of handling these as there's not a sensor created. Some ideas:
* add attributes to the operational state? Not sure it considered good practice to dynamically add and remove attributes from HA sensors?
* create sensor when it sees the alert come in and remove it when the alert clears?
* stick with the event scripts/templates above?

example below showing an event coming in and later being cleared on the debug log (level: alert, KEY is the event, and value is the eventpresentstate [present, off, confirmed])
```
2023-10-02 20:50:55.320 DEBUG (MainThread) [custom_components.home_connect_alt.home_connect_async.homeconnect] Received event from SSE stream: MessageEvent(type='EVENT', message='EVENT', data='{"haId":"SIEMENS-CT836LEB6-ABCDEFGHIJKL","items":[{"handling":"none","key":"ConsumerProducts.CoffeeMaker.Event.DripTrayFull","level":"alert","timestamp":1696276255,"value":"BSH.Common.EnumType.EventPresentState.Present"}]}', origin='https://api.home-connect.com', last_event_id='SIEMENS-CT836LEB6-ABCDEFGHIJKL')

2023-10-02 22:13:52.865 DEBUG (MainThread) [custom_components.home_connect_alt.home_connect_async.homeconnect] Received event from SSE stream: MessageEvent(type='EVENT', message='EVENT', data='{"haId":"SIEMENS-CT836LEB6-ABCDEFGHIJKL","items":[{"handling":"none","key":"ConsumerProducts.CoffeeMaker.Event.DripTrayFull","level":"alert","timestamp":1696281231,"value":"BSH.Common.EnumType.EventPresentState.Off"}]}', origin='https://api.home-connect.com', last_event_id='SIEMENS-CT836LEB6-ABCDEFGHIJKL')
```
